### PR TITLE
8344822: CDS BulkLoaderTest.java#dynamic fails with COH

### PR DESCRIPTION
**Pull request by iklam**
Creating a dynamic CDS archive (with `-XX:ArchiveClassesAtExit=<foo>.jsa`) requires that the current JVM process is using a static archive (which is usually the default CDS archive included in the JDK). The tests fails because the JVM is executed with `-XX:+UseCompactObjectHeaders`, but the JDK doesn't include a default CDS archive that's compatible with such VM options.

The fix is to detect for this situation, and create a temporary static archive to be used by the test.

Unfortunately this detection requires WhiteBox, so all test cases that use `CDSAppTester` with `@run ... DYNAMIC` need to be modified to add the appropriate jtreg tags.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344822](https://bugs.openjdk.org/browse/JDK-8344822): CDS BulkLoaderTest.java#dynamic fails with COH (**Bug** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Calvin Cheung](https://openjdk.org/census#ccheung) (@calvinccheung - **Reviewer**)

### Reviewers without OpenJDK IDs
 * @abdelhak-zaaim (no known openjdk.org user name / role)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22403/head:pull/22403` \
`$ git checkout pull/22403`

Update a local copy of the PR: \
`$ git checkout pull/22403` \
`$ git pull https://git.openjdk.org/jdk.git pull/22403/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22403`

View PR using the GUI difftool: \
`$ git pr show -t 22403`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22403.diff">https://git.openjdk.org/jdk/pull/22403.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22403#issuecomment-2502356463)
</details>

### CODE REVIEW

File: test/lib/jdk/test/lib/cds/CDSAppTester.java [hunk 1]
```diff
@@ -189,9 +192,37 @@ private OutputAnalyzer dumpStaticArchive() throws Exception {
...
+            return null;
+        } else {
+            // This current JVM is unable to use a default CDS archive, so let's create a temporary
+            // static archive to be used with -XX:ArchiveClassesAtExit.
+            File f = new File(tempBaseArchiveFile);
+            if (!f.exists()) {
+                CDSOptions opts = new CDSOptions();
+                opts.setArchiveName(tempBaseArchiveFile);
+                opts.addSuffix("-Djava.class.path=");
+                OutputAnalyzer out = CDSTestUtils.createArchive(opts);
```
Comment (dholmes-ora) [inline test/lib/jdk/test/lib/cds/CDSAppTester.java:215]: Just to be clear, this creates the right kind of archive based on whether +UCOH is used or not?

**Review by dholmes-ora — APPROVED** [review]
Seems reasonable.

Thanks

File: test/lib/jdk/test/lib/cds/CDSAppTester.java (hunk 1 above)
Comment (iklam) [reply test/lib/jdk/test/lib/cds/CDSAppTester.java:215]: Yes, `CDSTestUtils.createArchive()` uses `ProcessTools.createTestJavaProcessBuilder()` to launch a child process to create the archive, so it passes the test vm options to the child.

**Review by calvinccheung — APPROVED** [review]
LGTM

**Pull request closed without merging**